docs/articles/deployments.md

```diff
@@ -0,0 +1,237 @@
+- `StopApplication` - stops given service by killing its tasks and waiting for the kill confirmation
+- `RestartApplication` - used when modifying an existing service definition and restarting its tasks (using the new definition)
+
+Steps are executed sequentially, however actions within one step (should there be more than one) are executed in parallel. Once all steps are executed successfully, the deployment is finished and removed from the state.
+
+## Canceling a deployment
+Sometimes a deployment is "stuck" for some reason (see [troubleshooting](troubleshooting.md) for more information). In this case, it might be helpful to cancel the deployment, returning to the previous state. Let's cancel the above stuck deployment by sending a `DELETE /v2/deployments/3bcb15fc-737f-4a36-b185-2ae3a703239e` request. The response is `HTTP 200(ok)` but the payload might be somewhat a surprise:
+```
+{
+    "version": "2018-12-05T00:28:00.584Z",
+    "deploymentId": "342a6dc5-f871-4d10-a4ec-bbe21b431d58"
+}
+```
+We see that another `deploymentId` is returned. This is because internally canceling a deployment simply creats another deployment that reverts the steps of the original one.
+
+**Known caveats:** Marathon has a cap on the number of concurrently running deployments. This is controlled by `--max_running_deployments` [command line argument](command-line-arguments.md). In some rare occasions, one can have the maximum number of running deployments achieved so that canceling one is not possible (since a new deployment has to be created). In this case, canceling using `?force=true` parameter aka. [forced deployment](deployments.md#forced-deployments) should be used.
+
+## Forced deployments
+Let's get back to our `product/service/rails-app` service deployment that is now "stuck" with slightly exaggerated resource requirements. We realize that 1k CPUs is too much and decide to update the `product/service/rails-app` service definition with reasonable CPU requirements. However, when submitting new service definition, Marathon will respond with `HTTP 209 (conflict)`:
+```
+{
+    "message": "App is locked by one or more deployments. Override with the option '?force=true'",
+    "deployments": [
+        {
+            "id": "3aa7316a-7888-427b-9af3-4179a7791c79"
+        }
+    ]
+}
+```
+This is due to already existing (stuck) deployment that affects our service definition. In general, Marathon will only allow one deployment at a time affecting any service definition (or service definition group). Let's try to submit our corrected definition again, however, this time using `?force=true` parameter (`POST /v2/apps?force=true`). This time, the result is `HTTP 200(ok)` and we receive a new deployment Id back. Under the hood, Marathon will cancel the previous deployment and start the new one. The new deployment will reconcile existing tasks (if any), check their service definition version, kill old ones if necessary and start new tasks.
```

Comment:
```suggestion
This happened because already existing (stuck) deployment affects our service definition. In general, Marathon will only allow one deployment at a time affecting any service definition (or service definition group). Let's try to submit the corrected service definition again, however, this time using `?force=true` parameter (`POST /v2/apps?force=true`). Now, the result is `HTTP 200(ok)` and we receive a new deployment Id back. Under the hood, Marathon will cancel the previous deployment and start the new one. The new deployment will reconcile existing tasks (if any), check their service definition version, kill old tasks if necessary and start new tasks.
```